### ISSUE TITLE
Advance Search does not find anything it should

### ISSUE THREAD

**Issue by JonasW314** [texstudio-org/texstudio#3595]
I have a big LaTex Document which is structured like this:

\include{A Chapter}
Inside the chapter then

\input{Subsection}

Etc. 

If I'm now using the adavanced search this function is very unreliable. Usually it found only a few of the terms in the whole project I'm looking for. 

It works sometimes better if all files are open but it is also then not reliable. 

It seems to have more difficulties if a term is inside a footnote or a a command like \ref{} 

Textstudio Version 6.6.1
Windows 10
Miktex

If I could provide you with any further information I'm happy to do that.

**Comment by sunderme** [comment]
a complete example with all files and a step-by-step instructions on how to reproduce it are needed.

**Comment by JonasW314** [comment]
I maybe figured out what could be the problem after I rearead the manuel.

The programm seem to be able also to do some searching in files which were recently opend and then closed again but  not in files which were closed for a while.

This lead to some confusing by figuring out what is going on.

If I'm opening all files manually befor using "advanced search"  it seems to work. 

Could that be the problem? That the programm is not able to search in files which I didn't open for a while?

Even if it is clear which is the "root file"

==

I could provid you with a  complete example but that would mean to give you more or less full access to my whole research project. Which seem not be the best Idea. 

I could try to make an "artifical" example later on if that is necessary. But maybe I could already isolte the problem (see above)....

**Comment by sunderme** [comment]
have you disabled options/automatically load included files ?

**Comment by JonasW314** [comment]
Nope

I thought already if that could be the problem and checked it yesterday. The checkbox is on.

This is part of my confussion. 

![Einstellungen](https://github.com/texstudio-org/texstudio/assets/166590893/aecbd834-c73d-4621-9101-f262a4ca0fb5)

**Comment by JonasW314** [comment]
I also made explicit which is the "root file" by using the function of the programm or doing it manually by writing a magic comment. 

This seem not to make an difference.

**Comment by JonasW314** [comment]
What I thought first could be a problem is this:

![5 Files](https://github.com/texstudio-org/texstudio/assets/166590893/e786a50d-9993-4ad2-ba97-85cf0601ddf3)

As I thought maybe this limits the program to use 5 files. But as I changed it too 30 it does not seem to make an difference. But I could not figuring out for what this function actually are. 

So maybe this is irrelevant.

**Comment by sunderme** [comment]
That is indeed irrelevant.
View/list hidden documents & list open document  shows all the documents which txs has in memory (by automatically loading them or not purging them from memory when closing)

So, without any concrete information, how your document and set-up looks like, this all wild guess work.

**Comment by sunderme** [comment]
I can only offer that you send the documents directly to me (sunderme@gmx.de)

**Comment by JonasW314** [comment]
Thx I will consider that.

However it is already good to know that I don't expect the programm to do something which could not be done by the programm. 

If I got you right the programm should be able to load all files automatically during the "advanced search" right?

And the way I included them in my root document should not make a difference.

**Comment by sunderme** [comment]
When you open the root document, txs loads all included (and included in included) documents and scans them (references, command definitions etc). 
It is not necessary to set an explicit root/ magic comment.

**Comment by sunderme** [comment]
"options/editor/cache documents for faster reopening" needs to be disabled.
Direct side effect is that the documents are not completely loaded and are not searchable.
I may change that behaviour in future but for now that is one restriction.

**Comment by cdmckeon** [comment]
I just had the same problem as JoanW314 with the search function failing to search in included files I hadn't opened for a few days. Sunderme's comment that " "options/editor/cache documents for faster reopening" needs to be disabled" worked for me. I unchecked the box, closed and restarted texstudio, and searches brought up results in my included files I hadn't opened for awhile. I had also upped "max recent documents" in options/general" but with no result, either.  Thx to sunderme for that comment!
